homework/src/ba/bitcamp/homeworkweekend/Recursion.java

```diff
@@ -0,0 +1,35 @@
+package ba.bitcamp.homeworkweekend;
+
+
+
+
+public class Recursion {
+	/**
+	 * This method returns second number as first number exponent.
+	 * @param a The first number.
+	 * @param b	The second number.
+	 */
+	public static double getPower(double a, double b) {
+		if (b % 1 != 0)
+			return Math.pow(a, b);
+		if (b > 0) {
+			return getPower(a, b, true);
+		} else {
+			return 1 / getPower(a, b * -1, false);
+		}
+	}
+
+	private static double getPower(double a, double b, boolean isPositive) {
+		if (b == 0)
+			return 1;
+		return a * getPower(a, b - 1);
+	}
```

Comment:
Rekurzivna metoda bi se trebala pozvati direktno iz main metode i moze izgledati ovako:

	public static double getPower(double a, double b) {
		if (b == 0){
			return 1;
		} if (b > 0) {
			return a * getPower(a, b-1);
		} else {
// pozvat ce se samo jednom, ako je eksponent negativan broj
// castanje u double djeljenika ili djelitelja je bitno zbog nacina na koji se dijele brojevi u Javi
			return (double) 1 / a * getPower(a, (b*-1)-1);
		}
	}

Molim te da analiziras ovu metodu detaljno, jer su rekurzivne metode koncept koji nije jednostavno shvatiti. Javi se za sve nejasnoce i dodatna pitanja. Takodjer, trebao bi baciti exception u slucaju da su i baza i eksponent 0, jer je to nedefinisan slucaj.